beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/hive/HiveClient.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.cleanup.hive;
+
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.expediagroup.beekeeper.core.error.BeekeeperException;
+
+import com.hotels.hcommon.hive.metastore.client.api.CloseableMetaStoreClient;
+
+public class HiveClient {
+
+  private static final Logger log = LoggerFactory.getLogger(HiveClient.class);
+
+  private CloseableMetaStoreClient client;
+  private final boolean dryRunEnabled;
+
+  public HiveClient(CloseableMetaStoreClient client, boolean dryRunEnabled) {
+    this.client = client;
+    this.dryRunEnabled = dryRunEnabled;
+  }
+
+  /**
+   * Will drop the table from the database if it exists.
+   *
+   * @param databaseName
+   * @param tableName
+   */
+  public void dropTableIfExists(String databaseName, String tableName) {
```

Comment:
Minor, but you could rename this to just `dropTable` since `dropPartition` below also only does the drop if it exists and doesn't throw an error if not but the method isn't called `dropPartitionIfExists`.

Reply:
yeah I'm fine changing it back as well I asked for the rename to ...IfExists but now we got the trycatch(NoSuchObject..) it makes more sense to just call it dropTable again, sorry Jay.

Reply:
Okay I'll change the name 👍 

Do you also want me to change it back so it returns a boolean like the `dropPartition` method? Or fine as is?

Reply:
nah we don't need that and I still don't like that :)